Mathlib/Tactic/Linter/Header.lean

```diff
@@ -221,6 +221,19 @@ def broadImportsCheck (imports : Array Syntax)  : Array (Syntax × String) := Id
           make sure to benchmark it. If this is fine, feel free to allow this linter.")
   return output
 
+/-- Check the syntax `imports` for syntactically duplicate imports.
+The output is an array of `Syntax` atoms whose ranges are the import statements,
+and the embedded strings are the error message of the linter.
+-/
+def duplicateImportsCheck (imports : Array Syntax)  : Array Syntax := Id.run do
+  let mut output := #[]
+  let mut importsSoFar := #[]
+  for i in imports do
+    if importsSoFar.contains i then
+      output := output.push (mkAtomFrom i s!"Duplicate imports: '{i}' already imported")
+    else importsSoFar := importsSoFar.push i
+  return output
```

Comment:
```suggestion
def duplicateImportsCheck (imports : Array Syntax) : CommandElabM Unit := do
  let mut importsSoFar := #[]
  for i in imports do
    if importsSoFar.contains i then
      Linter.logLint linter.style.header i m!"Duplicate imports: '{i}' already imported"
    else importsSoFar := importsSoFar.push i
```
what do you think of making the function do the reporting, rather than accumulating an array and then looping over the array again later on?

Reply:
A similar refactor could apply to the rest of the linter: when I wrote it, I was mostly concerned with not modifying existing code, but for the current format it probably makes more sense to emit warnings on the spot, rather than accumulating them and then logging them all again later.